fixd.js

```diff
@@ -1,14 +1,55 @@
-const fs = require('fs')
-const path = require('path')
-const fixd = require('./src/index')
+'use strict'
+const merge = require('lodash.merge')
 
-// Auto-load built-in plugins
-try {
-  fs.readdirSync(path.resolve(__dirname, './src/plugins')).forEach((f) => {
-    fixd.use(f.replace('.js', ''))
-  })
-} catch (e) {
-  throw new Error('Error loading built-in plugins', e.message)
+const fixd = {
+  /**
+   * Deeply freezes an object
+   * @param {*} obj The object to freeze
+   * @returns {*}
+   */
+  freeze: (obj) => {
+    Object.freeze(obj)
+    Object.getOwnPropertyNames(obj).forEach((prop) => {
+      if (obj.hasOwnProperty(prop) &&
+      obj[prop] !== null &&
+      (typeof obj[prop] === 'object' || typeof obj[prop] === 'function') &&
+      !Object.isFrozen(obj[prop])) {
+        fixd.freeze(obj[prop])
+      }
+    })
+    return obj
+  },
+  /**
+   * Add a fixture to the fixd library and freezes if not in namespace
+   * @param {String} name A string name to represent the fixture
+   * @param {*} value The fixture value
+   */
+  add: (name, val) => {
+    if (fixd[name]) {
+      throw new Error(`Cannot add fixture to already reserved namespace '${name}'.`)
+    }
+    fixd[name] = fixd.freeze({ $fixdVal: val })
+  },
+  /**
+   * Returns a fixture and allows for (option) modification
+   * @param {String} name The namespace of the fixture to mutate
+   * @param {Function} modifier A function to modify the fixture
+   */
+  create: (name, modifier) => {
+    if (typeof modifier !== 'function') {
+      throw new Error('Modifier must be a function')
+    }
+    return fixd.freeze(modifier(merge(Array.isArray(fixd[name].$fixdVal) ? [] : {}, fixd[name].$fixdVal)))
+  }
 }
 
-module.exports = fixd
+module.exports = new Proxy(fixd, {
```

Comment:
Yeah, it was a consideration, however, I feel like this is the perfect use-case, and while it could limit adoption, I'd prefer lean forward than back. I'm happy to get this in for now (for our use-cases) and worry about it should we get an issue.